src/Sylius/Behat/Context/Api/Shop/ProductVariantContext.php

```diff
@@ -105,10 +104,10 @@ public function iShouldSeeVariantIsDiscountedFromToWithPromotions(
 
         Assert::same($content['price'], $price);
         Assert::same($content['originalPrice'], $originalPrice);
-        foreach ($promotionsNames as $promotionName) {
-            $catalogPromotionCode = StringInflector::nameToCode($promotionName);
-            Assert::same($content['appliedPromotions'][$catalogPromotionCode]['translations']['en_US']['name'], $promotionName);
-            Assert::same($content['appliedPromotions'][$catalogPromotionCode]['translations']['en_US']['description'], $promotionName . ' description');
+        foreach ($content['appliedPromotions'] as $promotionIri) {
+            $catalogPromotionResponse = $this->client->showByIri($promotionIri);
+            $catalogPromotion = $this->responseChecker->getResponseContent($catalogPromotionResponse);
```

Comment:
imo you can merge it to single line:
```suggestion
            $catalogPromotion = $this->responseChecker->getResponseContent(
                 $this->client->showByIri($promotionIri)
             );
```